ui/components/app/snaps/snap-home-menu/snap-home-menu.js

```diff
@@ -0,0 +1,153 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import { useI18nContext } from '../../../../hooks/useI18nContext';
+import {
+  Box,
+  ButtonIcon,
+  ButtonIconSize,
+  IconName,
+  Popover,
+  PopoverPosition,
+  PopoverRole,
+  Text,
+} from '../../../component-library';
+import {
+  AlignItems,
+  Display,
+  FlexDirection,
+  JustifyContent,
+  TextColor,
+  TextVariant,
+} from '../../../../helpers/constants/design-system';
+import { SnapMetadataModal } from '../snap-metadata-modal';
+
+export default function SnapHomeMenu({
+  snapId,
+  isSettingsAvailable,
+  onSettingsClick,
+  onRemoveClick,
+}) {
+  const t = useI18nContext();
+
+  const [menuReferenceElement, setMenuReferenceElement] = useState();
+
+  const [isMenuOpen, setIsMenuOpen] = useState(false);
+
+  const [isSnapMetadataModalOpen, setIsSnapMetadataModalOpen] = useState(false);
+
+  const handleMenuClick = (menuItem) => {
+    switch (menuItem) {
+      case 'settings':
+        onSettingsClick();
+        break;
+      case 'details':
+        setIsSnapMetadataModalOpen(true);
+        break;
+      case 'remove':
+        onRemoveClick();
+        break;
+      default:
+        break;
+    }
+    setIsMenuOpen(false);
+  };
+
+  const handleMenuOpen = () => {
+    setIsMenuOpen(!isMenuOpen);
+  };
+
+  const setMenuRef = (ref) => {
+    setMenuReferenceElement(ref);
+  };
+
+  const closeMenu = () => {
+    setIsMenuOpen(false);
+  };
+
+  const closeSnapMetadataModal = () => setIsSnapMetadataModalOpen(false);
+
+  return (
+    <>
+      {isSnapMetadataModalOpen && (
+        <SnapMetadataModal
+          snapId={snapId}
+          isOpen={isSnapMetadataModalOpen}
+          onClose={closeSnapMetadataModal}
+        />
+      )}
+      <Box display={Display.Flex}>
+        <Box
+          display={Display.Flex}
+          justifyContent={JustifyContent.center}
+          alignItems={AlignItems.center}
+        >
+          <ButtonIcon
+            iconName={IconName.MoreVertical}
+            data-testid="snap-home-menu-button"
+            ariaLabel={t('snapHomeMenu')}
+            onClick={() => {
+              handleMenuOpen();
+            }}
+            size={ButtonIconSize.Md}
+            ref={setMenuRef}
+          />
+        </Box>
+        <Popover
+          referenceElement={menuReferenceElement}
+          isOpen={isMenuOpen}
+          position={PopoverPosition.Bottom}
```

Comment:
Can we do `BottomStart` or `BottomEnd` like in the design ?
![image](https://github.com/MetaMask/metamask-extension/assets/13910212/75a8d066-fb3b-40d7-a70d-7b3878ecc1fd)

Reply:
Changed to `BottomEnd` and made some adjustments with the offset so it looks closer to the design.